app/utilities/decimal_places.py

```diff
@@ -0,0 +1,113 @@
+from decimal import Decimal
+from typing import Literal, TypeAlias
+
+from babel import Locale, numbers, units
+
+UnitLengthType: TypeAlias = Literal["short", "long", "narrow"]
+
+
+def custom_format_decimal(value: int | Decimal | float, locale: Locale | str) -> str:
+    """
+    This function provides a wrapper for the numbers `format_decimal` method, generating the
+    number format (including the desired number of decimals), based on the value entered by the user and
+    the locale.
+    """
+    number_format = get_number_format(value, locale)
+
+    return numbers.format_decimal(
+        value,
+        format=number_format,
+        locale=locale,
+    )
+
+
+def custom_format_currency(
+    value: float | Decimal,
+    currency: str,
+    locale: Locale | str,
+    decimal_limit: int | None = None,
+) -> str:
+    """
+    This function provides a wrapper for the numbers `format_currency` method, generating the
+    number format (including the desired number of decimals).
+
+    The number of decimals displayed is based on the value entered by the user, the decimal limit set in the schema
+    and the locale.
+    """
+    decimal_places = len(str(value).split(".")[1]) if "." in str(value) else 0
+
+    # get locale pattern
+    parsed_locale = Locale.parse(locale)
+    number_format = parsed_locale.currency_formats["standard"]
+
+    # If set, the number of decimals displayed is limited based on the value of the `decimal_limit` parameter.
+    # If the number of decimal places entered by the user is less than the `decimal_limit` then we should display the
+    # number of decimals as entered by the user. Otherwise, we should display the number of decimals as entered by the user.
+    if decimal_limit:
```

Comment:
It gets confusing because no decimal limit defaults to whatever the user enters, but I'm wondering if thats what we should be doing - if there is no decimal limit, should currency not default to 2?

Reply:
Updated with the code from the above comment, but simplifying further with `currency_digits=decimal_limit is None,` doesn't seem to work.

Reply:
Ahh... yeah if you have 100, and decimal places 6, it would not change it to £100.00 in that instance, so I see why the 2 was needed. Since every currency format uses 2 for frac_prec[1] this seems safe

Reply:
Will discuss with Donna tomorrow however food for thought: Author by default sets decimals to 0, therefore, a normal currency field will not allow decimal and display will use no decimal places since schema limit is 0 by default. Therefore, we probably should treat not defined as 0 and only use the locale limit of 2 if the user entered 1 decimal place. So £1.1 would display as £1.10 unless schema limit is 1.